### ISSUE TITLE
Ignore .cache files when evaluating files in an inventory directory

### ISSUE THREAD

**Issue by geerlingguy** [ansible/ansible#9523]
##### Issue Type:

Feature Idea
##### Ansible Version:

`ansible 1.7.2`
##### Environment:

OSX Yosemite provisioning host, multiple managed hosts
##### Summary:

It would be nice if Ansible would ignore .cache files in mixed inventory directories. The Digital Ocean dynamic inventory script (don't know about others right now) creates a .cache file after it runs, in the same directory as the script. The next time Ansible is run with the inventory directory specified, I get the error:

```
ERROR: Invalid ini entry: { - need more than 1 value to unpack
```

...because Ansible tries parsing the .cache file as an inventory file.
##### Steps To Reproduce:
1. Create an inventory directory, and place the digital_ocean.py dynamic inventory script inside.
2. Run something like `ansible-playbook main.yml -i inventory/` (where `inventory/` contains the dynamic script and any other relevant inventories).
##### Expected Results:

Ansible ignores `.cache` files.
##### Actual Results:

Error:

```
$ ansible-playbook main.yml -i inventory/
ERROR: Invalid ini entry: { - need more than 1 value to unpack
```

**Comment by geerlingguy** [comment]
Note that this isn't an issue normally... but when I'm developing new playbooks/functionality and using the digital_ocean.py script directly, the .cache file keeps popping up in that directory (it defaults to saving in the current directory).

**Comment by alobbs** [comment]
I haven't been able to reproduce this one with the current devel branch.

**Comment by jimi-c** [comment]
@geerlingguy is this one still an issue? Really this should have been classified as a bug at the time.

**Comment by geerlingguy** [comment]
I'll try to reproduce. I think it was with one of the examples in Ansible for DevOps, and I need to update that example anyways, so I might as well do some extra testing on it now!

**Comment by bcoca** [comment]
any hidden file is skipped (starts with .) fixed in 69db160e2fff407ca89340ed5728d8fda89b65b2

**Comment by geerlingguy** [comment]
Well, there you have it! Thanks.
